BackupDatabase-Recipe.rb

```diff
@@ -0,0 +1,76 @@
+require 'nokogiri'
+
+def get_database_credentials_for_magento
+  configXMLFile = Nokogiri::XML(File.open("#{current_release}#{app_path}/app/etc/local.xml"))
+  dbCredentials = Hash.new
+  ["host", "username", "password", "dbname"].each do |credential|
+    dbCredentials[credential] = configXMLFile.xpath("//config//global//resources//default_setup//#{credential}/text()").text
+  end
+  dbCredentials
+end
+
+def get_database_credentials_for_cakephp
+  configFile = File.open("#{current_release}#{app_path}/Config/database.php").read
+
+  rowCredentials= configFile.scan(/'(?<key>\w+)'\s=>\s'(?<value>\w+)'/)
+  dbCredentials = Hash.new
+
+  mapping = {"login" => "username", "database" => "dbname"}
+  rowCredentials.each do |key,value|
+    credentialKey = mapping[key].nil? ? key : mapping[key]
+    dbCredentials[credentialKey] = value
+  end
+
+  dbCredentials
+end
+
+def is_cakephp_project
+  exists?(:cake_migrations)
+end
+
+namespace :database do
+
+  desc <<-DESC
+		Generate a backup for database project.
+  DESC
+  task :backup, :roles => :db, :only => { :primary => true } do
+    filename = "#{application}.dump.sql.gz"
+    file = "/tmp/#{filename}"
+    on_rollback { delete file }
+
+    logger.info("Fetching database credentials from project")
+    dbCredentials = is_cakephp_project ? get_database_credentials_for_cakephp : get_database_credentials_for_magento
+
+    logger.info("Dumping database")
+    if dbCredentials.length > 0
+      run "mysqldump  --add-drop-table --extended-insert --force -u #{dbCredentials["username"]} --password=#{dbCredentials["password"]} #{dbCredentials["dbname"]} -h #{dbCredentials["host"]} | gzip > #{file}"  do |ch, stream, data|
```

Comment:
It is generally a good practice to have binary files configurable with good defaults. For instance on systems one may not have mysqldump in the path, so it should read `/usr/bin/mysqldump`.

Suggestion: `_cset(:mysqldump_bin) { "mysqldump" }`

See https://github.com/occitech/capistrano-recipes/blob/e69484a060ea64b6b5b6dda94600d4f2ec63110a/CakePHP2x-Recipe.rb#L20 for example

Same thing for other binaries